mmf/modules/encoders.py

```diff
@@ -12,14 +15,44 @@
 from mmf.utils.download import download_pretrained_model
 from mmf.utils.file_io import PathManager
 from mmf.utils.general import get_absolute_path
-from omegaconf import OmegaConf
+from omegaconf import MISSING, OmegaConf
 from torch import nn
 from transformers.configuration_auto import AutoConfig
 from transformers.modeling_auto import AutoModel
 
 
+class ImageFeatureEncoderTypes(Enum):
+    default = "default"
+    identity = "identity"
+    projection = "projection"
+    frcnn_fc7 = "finetune_faster_rcnn_fpn_fc7"
+
+
+# SuperClass for encoder params
+@dataclass
+class EncoderParams:
+    pass
+
+
+@dataclass
+class EncoderConfig:
+    type: str = MISSING
```

Comment:
Just curious, what does `MISSING` do compared to leaving it without any value? And if a value is required why not use `MISSING` everywhere in all the dataclass additions in this PR?

Reply:
MISSING means that value will need to be passed by the user in all cases. I will review again to see if I am missing "MISSING" anywhere it should be present.